test/e2e/suite/conformance/certificates/suite.go

```diff
@@ -69,6 +62,12 @@ type Suite struct {
 	// certain features due to restrictions in their implementation.
 	UnsupportedFeatures FeatureSet
 
+	// ValidateCertificateChecks is a list of functions that check if the
+	// issued certificate is valid.
+	// This is defined per issuer as each issuer might have a specific
+	// set CA signing threats, like setting or removing subject fields
```

Comment:
I'm a bit concerned that if issuers override this, they will change the actual meaning of the tests/what is being checked in the tests.

In an ideal world, these test cases would test quite specific properties of the Certificate, and so something like this shouldn't be necessary. If a specific type of test then isn't possible, it'd disable that 'feature' using the `UnsupportedFeatures` field.

I'm not too sure if this is really the direction we should be going, at least not as an option within the `Suite`. It feels like we're trying to create one verification function to "rule them all" (i.e. used by _every_ test case), which doesn't solve the underlying issue we have right now with the `WaitForCertificateReady` function (which is that it is opaque and not clear whether it is properly checking the right things in the right cases)

Reply:
Would it be a solution to replace this slice with a function that returns checks based on the feature set of the issuer?